ghcide/src/Development/IDE/GHC/ExactPrint.hs

```diff
@@ -292,8 +296,8 @@ graftExprWithM dst trans = Graft $ \dflags a -> do
                         case mval of
                             Just val' -> do
                                 (anns, val'') <-
-                                    hoistTransform (either Fail.fail pure) $
-                                        annotate dflags needs_space $ mk_parens val'
+                                    hoistTransform (either Fail.fail pure)
+                                        (annotate @(HsExpr GhcPs) dflags needs_space (mk_parens val'))
```

Comment:
A type annotation is needed here to prevent the type checker from diverging, but no idea why

Reply:
Weird. Sounds like a bug in GHC to me; since `mval` is very definitely an `LHsExpr GhcPs`.